webui/portal/src/main/java/org/exoplatform/portal/application/DefaultRequestHandler.java

```diff
@@ -68,21 +63,24 @@ public boolean execute(ControllerContext context) throws Exception {
 
         if (StringUtils.isBlank(defaultUri)) {
           String defaultPortal = configService.getDefaultPortal();
-          List<String> portalNames = configService.getSiteNames(SiteType.PORTAL, 0, 10);
-          boolean emptyPortalList = portalNames == null || portalNames.isEmpty();
-          boolean canAccessDefaultPortal = portalNames != null && portalNames.contains(defaultPortal);
-          if (!emptyPortalList && !canAccessDefaultPortal) {
-            defaultPortal = portalNames.get(0);
-          } else if (emptyPortalList) {
+          SiteFilter siteFilter = new SiteFilter();
+          siteFilter.setExcludedSiteName(configService.getGlobalPortal());
+          siteFilter.setSiteType(SiteType.PORTAL);
+          siteFilter.setSortByDisplayOrder(true);
+          siteFilter.setFilterByDisplayed(true);
+          siteFilter.setDisplayed(true);
+          List<PortalConfig> portalConfigList = configService.getSites(siteFilter);
```

Comment:
This SQL query has to be indexed in Database, which is highly used + cached if possible to not having to request a database every display of a page while the number of sites doesn't change too much.

Reply:
we will indexe the PORTAL_SITES fields as recommended [here ](https://github.com/Meeds-io/gatein-portal/pull/647#discussion_r1347320305)

Reply:
ok, it's a first step, coupled with a `304 HTTP response` this would be sufficient to have an efficient performant page rendering.

Reply:
yes the 304 will be for the [getSites rest ](https://github.com/Meeds-io/social/pull/2747#discussion_r1347407382)